Sources/Commands/UserToolchain.swift

```diff
@@ -57,21 +57,19 @@ public struct UserToolchain: Toolchain, ManifestResourceProvider {
         let envSearchPaths = Utility.getEnvSearchPaths(
             pathString: getenv("PATH"), currentWorkingDirectory: currentWorkingDirectory)
 
-        func lookup(env: String) -> AbsolutePath? {
+        func lookup(fromEnv: String) -> AbsolutePath? {
             return Utility.lookupExecutablePath(
-                inEnvValue: getenv(env),
+                filename: getenv(fromEnv),
                 searchPaths: envSearchPaths)
         }
 
         libDir = binDir.parentDirectory.appending(components: "lib", "swift", "pm")
 
         // First look in env and then in bin dir.
-        swiftCompiler = lookup(env: "SWIFT_EXEC") ?? binDir.appending(component: "swiftc")
+        swiftCompiler = lookup(fromEnv: "SWIFT_EXEC") ?? binDir.appending(component: "swiftc")
         
-        // Check that it's valid in the file system.
-        // FIXME: We should also check that it resolves to an executable file
-        //        (it could be a symlink to such as file).
-        guard localFileSystem.exists(swiftCompiler) else {
+        // Check that it's valid in the file system and executable.
+        guard localFileSystem.isExecutableFile(swiftCompiler) else {
```

Comment:
Ah I see, thats a fair categorization. I was thinking about this PR as a feature addition to Process class (which I still do, see previous comment).